src/engine/Interfaces/LifecycleEvents.ts

```diff
@@ -88,11 +88,18 @@ export interface CanInitialize {
   off(eventName: Events.initialize, handler?: (event: Events.InitializeEvent<any>) => void): void;
 }
 
-export interface CanActivate {
+export interface SceneActivationContext<TData = undefined> {
```

Comment:
@eonarheim I named this TData as per the proposal in #1548, but I don't think we follow the `<TSomething>` naming convention with generics anywhere else (it seems to usually be `<T>`). I feel like either `<Data>` or `<T>` would be more consistent here.